runtime/near-vm-logic/src/logic.rs

```diff
@@ -64,14 +64,43 @@ pub struct VMLogic<'a> {
     total_log_length: u64,
 }
 
+#[derive(Clone)]
+pub struct InternalVMState {
```

Comment:
Perhaps we need to meet to discuss this in more detail. But my current working solution (since this is my first real rust project), instantiates a new VM for each api call.  So this allows the state to be saved and restored between invocations. It also allows the test interface to save and restore the state: see here: https://github.com/nearprotocol/wasm-mock-vm/blob/dca54c5ba9f2bccdac472c4926608e8140424239/assembly/__tests__/runtime.spec.ts#L36

Reply:
I don't think `InternalVMState` needs to be in defined in `near-vm-logic`. All instrumentation that we need to compile VMLogic should live in `wasm-mock-vm`.  Also, I looked at https://github.com/nearprotocol/wasm-mock-vm/blob/master/src/mock/vm_logic.rs#L72 and I am not sure we need `InternalVMState` at all, it basically `VMLogic` but with references replaced with cloning the only place it can be useful is to save/restore the state of `VMLogic`, but it is not needed anyway. Every time we run a test we can create `VMLogic` from the scratch like we do here: https://github.com/nearprotocol/near-sdk-rs/blob/master/near-bindgen/src/lib.rs#L34.

The final goal is to compile `nearcore/runtime/runtime` and `nearcore/core/store` crates and the state will be persisted on the trie so we won't need to store it through a temporary structure.